src/Sulu/Bundle/PreviewBundle/Preview/Renderer/PreviewRenderer.php

```diff
@@ -264,10 +264,10 @@ private function createPortalInformation($object, $id, $webspaceKey, $locale)
         $portal->setLocalizations([$localization]);
         $portal->setDefaultLocalization($localization);
 
-        $environment = new Environment();
-        $url = new Url($domain, $this->environment);
-        $environment->setUrls([$url]);
-        $portal->setEnvironments([$environment]);
+        $environment = new Environment($this->environment);
+        $environment->addUrl(new Url($domain));
+
+        $portal->addEnvironment($environment);
```

Comment:
Using the `addUrl` and `addEnvironment` calls here propagates the environment through to the url and the portal. It's still quite a BC layer. But maybe we can reduce this in 3.0